pkg/kubectl/cmd/delete/delete.go

```diff
@@ -206,11 +206,6 @@ func (o *DeleteOptions) Validate(cmd *cobra.Command) error {
 		return fmt.Errorf("cannot set --all and --field-selector at the same time")
 	}
 
-	if o.GracePeriod == 0 && !o.ForceDeletion && !o.WaitForDeletion {
```

Comment:
Why are you removing this? This was added intentionally, iirc, please leave it here.

Reply:
@soltysh look at the PR description, it can't happen because of [this block](https://github.com/kubernetes/kubernetes/blob/797f0ac660e60701e47c38ce2f2e0a21c8f19ec2/pkg/kubectl/cmd/delete/delete.go#L159-L163)

Reply:
Sorry, I missed the description, you're right, this is ok.